### PR TITLE
added Fetch All to DB API

### PR DESCRIPTION
**Pull request by albertlast**
added $smcFunc['db_fetch_all'] to mysql and pg
could be use full where you try to proceed many row sets at once like the time topic: https://github.com/SimpleMachines/SMF2.1/pull/4165

### CODE REVIEW

File: Sources/Subs-Db-mysql.php [hunk 1]
```diff
@@ -59,6 +59,7 @@ function smf_db_initiate($db_server, $db_name, $db_user, $db_passwd, $db_prefix,
 			'db_is_resource'            => 'smf_is_resource',
 			'db_mb4'                    => false,
 			'db_ping'                   => 'mysqli_ping',
+			'db_fetch_all'				=> 'smf_db_fetch_all',
```
Comment (colinschoen) [inline Sources/Subs-Db-mysql.php]: The tabbing looks off.

**Review by colinschoen — CHANGES_REQUESTED** [review]

**Comment by sbulen** [comment]
Do we want the @s in the return statements?

I don't think we want errors ignored.  Also, none of the other simply called functions use it in that fashion, e.g., mysqli_get_server_info().  We should be consistent.

**Comment by albertlast** [comment]
About the @ i see no consistency,
In Subs-Db-postgresql.php every fetch methode use @ 
and we don't ignore the error because of https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Load.php#L3047
when you like i can do in pg with @ and in mysql without,
but i don't see any reason for this...

**Comment by Oldiesmann** [comment]
I'm not sure why we're suppressing errors for PostgreSQL, but we aren't for MySQL at all, so at the very least be consistent that way. We should look into why we are suppressing errors in PostgreSQL for the fetch functions though.

**Pull request merged**